src/gui/core/canvas_private.hpp

```diff
@@ -303,7 +303,7 @@ class text_shape : public rect_bounded_shape
 	wfl::formula actions_formula_;
 
 	/** Any custom Pango text attributes. */
-	font::attribute_list text_attributes_;
+	config::const_child_itors text_attributes_;
```

Comment:
This line in particular is dangerous, since it forces the backing config to continue existing as long as the shape does. I believe *in practice* that is the case, but it's a hidden precondition that can lead to crashes if you're at some point you end up accessing an iterator into a since-destroyed config. Thinking of how best to resolve this.